starlite/testing/client/base.py

```diff
@@ -0,0 +1,144 @@
+        "root_path": "",
+        "scheme": "http",
+        "query_string": b"",
+        "client": ("testclient", 50000),
+        "server": ("testserver", 80),
+        "method": "GET",
+        "http_version": "1.1",
+        "extensions": {"http.response.template": {}},
+        "app": app,
+        "state": {},
+        "path_params": {},
+        "route_handler": None,
+        "_cookies": cookies,
+    }
+    return ASGIConnection[Any, Any, Any](
+        scope=scope,  # type: ignore[arg-type]
+    )
+
+
+class BaseTestClient(Generic[T]):
+    __test__ = False
+    blocking_portal: "BlockingPortal"
+
+    __slots__ = (
+        "app",
+        "base_url",
+        "backend",
+        "backend_options",
+        "session_config",
+        "_session_backend",
+        "cookies",
+    )
+
+    def __init__(
+        self,
+        app: T,
+        base_url: str = "http://testserver.local",
+        backend: AnyIOBackend = "asyncio",
+        backend_options: Optional[Dict[str, Any]] = None,
+        session_config: Optional["BaseBackendConfig"] = None,
+        cookies: Optional["CookieTypes"] = None,
+    ):
+        if "." not in base_url:
+            warnings.warn(
+                f"The base_url {base_url!r} might cause issues. Try adding a domain name such as .local: "
+                f"'{base_url}.local'",
+                UserWarning,
+            )
+        self._session_backend: Optional["BaseSessionBackend"] = None
+        if session_config:
+            self._session_backend = session_config._backend_class(config=session_config)
+        self.app = app
+        self.base_url = base_url
+        self.backend = backend
+        self.backend_options = backend_options
+        self.cookies = cookies
+
+    @property
+    def session_backend(self) -> "BaseSessionBackend":
+        if not self._session_backend:
+            raise ImproperlyConfiguredException(
+                "Session has not been initialized for this TestClient instance. You can"
+                "do so by passing a configuration object to TestClient: TestClient(app=app, session_config=...)"
+            )
+        return self._session_backend
+
+    @contextmanager
+    def portal(self) -> Generator["BlockingPortal", None, None]:
+        """Get a BlockingPortal.
+
+        Returns:
+            A contextmanager for a BlockingPortal.
+        """
+        if hasattr(self, "blocking_portal"):
+            yield self.blocking_portal
+        else:
+            with start_blocking_portal(backend=self.backend, backend_options=self.backend_options) as portal:
+                yield portal
+
+    @staticmethod
+    def _create_session_cookies(backend: "CookieBackend", data: Dict[str, Any]) -> Dict[str, str]:
+        encoded_data = backend.dump_data(data=data)
+        return {cookie.key: cast("str", cookie.value) for cookie in backend._create_session_cookies(encoded_data)}
+
+    async def _set_session_data_async(self, data: Dict[str, Any]) -> None:
```

Comment:
```suggestion
    async def _set_session_data(self, data: Dict[str, Any]) -> None:
```

I don't mean to nitpick, but since there's no `_set_session_data_sync`, the `_async` suffix is a bit superfluous.